tests/foreman/cli/test_product.py

```diff
@@ -190,3 +192,62 @@ def test_negative_create_with_label(self):
                         u'name': product_name,
                         u'organization-id': self.org['id'],
                     })
+
+    @tier2
+    def test_product_list_with_default_settings(self):
+        """Listing product of an organization apart from default organization using hammer
+         does not return output if a defaults settings are applied on org.
+
+        :id: d5c5edac-b19c-4277-92fe-28d9b9fa43ef
+
+        :BZ: 1745575
+
+        :expectedresults: product/reporsitory list should work as expected.
+
+        """
+        default_product_name = gen_string('alpha')
+        non_default_product_name = gen_string('alpha')
+        default_org = make_org()
+        non_default_org = make_org()
+        default_product = make_product({
+            u'name': default_product_name,
+            u'organization-id': default_org['id'],
+        })
+        non_default_product = make_product({
+            u'name': non_default_product_name,
+            u'organization-id': non_default_org['id'],
+        })
+        for product in (default_product, non_default_product):
+            make_repository({
+                'product-id': product['id'],
+                'url': FAKE_0_YUM_REPO,
+            })
+            Product.synchronize({
+                'id': product['id'],
+            })
+            packages = Package.list({'product-id': product['id']})
+            self.assertEqual(len(packages), FAKE_0_YUM_REPO_PACKAGES_COUNT)
```

Comment:
I'm not sure why we are synchronizing and checking packages. The bz and the following asserts are just about listing the prods and repos. So just creating the entities should do, potential for speedup. Syncing is well tested elsewhere

Reply:
I think no customer will only just create the repository, they will sync those repos ultimately. I'm just trying to think like the customer while designing this scenario.